Rules Based Decision Engine/src/models/Entity.java

```diff
@@ -0,0 +1,36 @@
+package models;
+
+import java.util.HashMap;
+
+/**
+ * An Entity is a collection of Data objects.
+ *
+ * An Entity represents a "thing" like, for example, a submarine.
+ * The submarine may have a temperature, latitude, and longitude.
+ * The HashMap field <code>map</code> is a collection of the above datapoints as Data objects.
+ *
+ * @author Michael Crinite
+ */
+public class Entity
+{
+    private String name;
+    private HashMap<String, Data> map;
+
```

Comment:
Add constructor with string and HM <Data> arguments. 
Data will be normalized outside this class prior to the Constructor being called. normalizeData function will be taken care of already so assume data is passed in the constructor is valid.